rskj-core/src/main/java/co/rsk/cli/tools/DbMigrate.java

```diff
@@ -0,0 +1,211 @@
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package co.rsk.cli.tools;
+
+import co.rsk.RskContext;
+import co.rsk.cli.CliToolRskContextAware;
+import org.ethereum.datasource.DbKind;
+import org.ethereum.datasource.KeyValueDataSource;
+import org.ethereum.util.FileUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nonnull;
+import java.io.File;
+import java.io.IOException;
+import java.lang.invoke.MethodHandles;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.stream.Stream;
+
+/**
+ * The entry point for db migration CLI tool
+ * This is an experimental/unsupported tool
+ *
+ * Required cli args:
+ * - args[0] - database target where we are going to insert the information from the current selected database.
+ *
+ * We do support the migrations between the following databases:
+ * - LevelDb (leveldb as argument)
+ * - RocksDb (rocksdb as argument)
+ */
+public class DbMigrate extends CliToolRskContextAware {
+    private static final Logger logger = LoggerFactory.getLogger(DbMigrate.class);
+    private static final String NODE_ID_FILE = "nodeId.properties";
+
+    private static class DbInformation {
+        private final KeyValueDataSource keyValueDataSource;
+        private final String indexPath;
+
+        public DbInformation(KeyValueDataSource keyValueDataSource, String indexPath) {
+            this.keyValueDataSource = keyValueDataSource;
+            this.indexPath = indexPath;
+        }
+
+        public KeyValueDataSource getKeyValueDataSource() {
+            return keyValueDataSource;
+        }
+
+        public String getIndexPath() {
+            return indexPath;
+        }
+    }
+
+    private static class DbMigrationInformation {
+        private final DbInformation targetDbInformation;
+        private final DbInformation sourceDbInformation;
+
+        public DbMigrationInformation(DbInformation targetDbInformation, DbInformation sourceDbInformation) {
+            this.targetDbInformation = targetDbInformation;
+            this.sourceDbInformation = sourceDbInformation;
+        }
+
+        public DbInformation getTargetDbInformation() {
+            return targetDbInformation;
+        }
+
+        public DbInformation getSourceDbInformation() {
+            return sourceDbInformation;
+        }
+    }
+
+
+    public static void main(String[] args) {
+        create(MethodHandles.lookup().lookupClass()).execute(args);
+    }
+
+    @SuppressWarnings("unused")
+    public DbMigrate() {
+    }
+
+    @Override
+    protected void onExecute(@Nonnull String[] args, @Nonnull RskContext ctx) throws IOException {
+        logger.info("Starting db migration...");
+
+        DbKind sourceDbKind = ctx.getRskSystemProperties().databaseKind();
+        DbKind targetDbKind = DbKind.ofName(args[0]);
```

Comment:
can we handle the case when a user doesn't provide any args in a command line and probably print an appropriate message? Looks like, as of now, it would simply crash with the standard index out of bounds error..

Reply:
Done